lib/waffle/storage/google/url_v2.ex

```diff
@@ -79,18 +79,9 @@ defmodule Waffle.Storage.Google.UrlV2 do
 
   @spec build_signed_url(Types.definition, String.t, Keyword.t) :: String.t
   defp build_signed_url(definition, path, options) do
-    {:ok, client_id} = Goth.Config.get(:client_email)
-
-    expiration = System.os_time(:second) + expiry(options)
-
-    signature = definition
-    |> build_path(path)
-    |> canonical_request(expiration)
-    |> sign_request()
-
-    base_url = build_url(definition, path)
-
-    "#{base_url}?GoogleAccessId=#{client_id}&Expires=#{expiration}&Signature=#{signature}"
+    client = GcsSignedUrl.Client.load_from_file(definition.service_account_path())
```

Comment:
instad of using load from file and have this extra configuration, you can use

definition.goth_module().Config.get(:client_email) and use the strategy as shown in: https://github.com/alexandrubagu/gcs_signed_url#example

with oauth config

Reply:
Goth.Config module is deprecated. We shouldn't use it anymore.
https://github.com/peburrows/goth/blob/master/CHANGELOG.md#v130-2022-06-29